cmd/keys.go

```diff
@@ -131,61 +132,111 @@ func keysRestoreCmd(a *appState) *cobra.Command {
 			}
 
-			if coinType < 0 {
-				if ccp, ok := chain.ChainProvider.(*cosmos.CosmosProvider); ok && ccp.PCfg.Slip44 != nil {
-					coinType = int32(*ccp.PCfg.Slip44)
-				} else {
-					coinType = int32(defaultCoinType)
+			if !restoreAll {
+
+				keyName := args[1]
+				chain, ok := a.config.Chains[args[0]]
+				if !ok {
+					return errChainNotFound(args[0])
 				}
-			}
 
-			algo, err := cmdFlags.GetString(flagAlgo)
-			if err != nil {
-				return err
-			}
+				if chain.ChainProvider.KeyExists(keyName) {
+					return errKeyExists(keyName)
+				}
 
-			if algo == "" {
-				if ccp, ok := chain.ChainProvider.(*cosmos.CosmosProvider); ok {
-					algo = ccp.PCfg.SigningAlgorithm
-				} else {
-					algo = string(hd.Secp256k1Type)
+				coinType, err := cmdFlags.GetInt32(flagCoinType)
+				if err != nil {
+					return err
 				}
-			}
 
-			address, err := chain.ChainProvider.RestoreKey(keyName, args[2], uint32(coinType), algo)
-			if err != nil {
-				return err
-			}
+				if coinType < 0 {
+					if ccp, ok := chain.ChainProvider.(*cosmos.CosmosProvider); ok && ccp.PCfg.Slip44 != nil {
+						coinType = int32(*ccp.PCfg.Slip44)
+					} else {
+						coinType = int32(defaultCoinType)
+					}
+				}
 
-			fmt.Fprintln(cmd.OutOrStdout(), address)
+				algo, err := cmdFlags.GetString(flagAlgo)
+				if err != nil {
+					return err
+				}
+
+				if algo == "" {
+					if ccp, ok := chain.ChainProvider.(*cosmos.CosmosProvider); ok {
+						algo = ccp.PCfg.SigningAlgorithm
+					} else {
+						algo = string(hd.Secp256k1Type)
+					}
+				}
+
+				address, err := chain.ChainProvider.RestoreKey(keyName, args[2], uint32(coinType), algo)
+				if err != nil {
+					return err
+				}
+				fmt.Fprintln(cmd.OutOrStdout(), address)
+				return nil
+
+			} else {
+
+				chains := a.config.Chains
+				keyName := args[0]
+				for _, c := range chains {
+
+					coinType, err := cmdFlags.GetInt32(flagCoinType)
+					if err != nil {
+						return err
+					}
+
+					if coinType < 0 {
+						if ccp, ok := c.ChainProvider.(*cosmos.CosmosProvider); ok && ccp.PCfg.Slip44 != nil {
+							coinType = int32(*ccp.PCfg.Slip44)
+						} else {
+							coinType = int32(defaultCoinType)
+						}
+					}
+
+					algo, err := cmdFlags.GetString(flagAlgo)
+					if err != nil {
+						return err
+					}
+
+					if algo == "" {
+						if ccp, ok := c.ChainProvider.(*cosmos.CosmosProvider); ok {
+							algo = ccp.PCfg.SigningAlgorithm
+						} else {
+							algo = string(hd.Secp256k1Type)
+						}
+					}
+
+					addresses, err := c.ChainProvider.RestoreKey(keyName, args[1], uint32(coinType), algo)
+					if err != nil {
+						return err
+					}
+
+					fmt.Fprintln(cmd.OutOrStdout(), addresses)
+				}
+			}
 			return nil
 		},
 	}
 	cmd.Flags().Int32(flagCoinType, -1, "coin type number for HD derivation")
 	cmd.Flags().String(flagAlgo, "", "signing algorithm for key (secp256k1, sr25519)")
+	cmd.Flags().Bool(flagRestoreAll, false, "restoring keys of all chains with single mnemonic")
```

Comment:
not to be too pedantic but can we change the description a bit to be something like, `restores keys for all configured chains with a single mnemonic`